spec/controllers/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe V1::UsersController, type: :controller do
+  include_context 'authentication'
+
+  let!(:user2) { FactoryGirl.create(:user) }
+
+  describe 'GET #find' do
+    let(:action) do
+      proc do
+        get :find, ids: [user.id, user2.id]
+      end
+    end
+
+    it_behaves_like 'authenticated_action'
+
+    describe 'should return the users' do
+      before do
+        action.call
+      end
+
+      it { expect(JSON.parse(response.body)['users'][0]['id']).to eq(user.id.to_s) }
+      it { expect(JSON.parse(response.body)['users'][1]['id']).to eq(user2.id.to_s) }
+    end
+  end
+
+  describe 'GET #show' do
+    let(:action) do
+      proc do
+        get :show, id: user.id
+      end
+    end
+
+    it_behaves_like 'authenticated_action'
+
+    describe 'should return the user' do
+      before { action.call }
+
+      it { expect(JSON.parse(response.body)['user']['id']).to eq(user.id.to_s) }
+    end
+  end
+
+  describe 'GET #current' do
+    let(:action) do
+      proc do
+        get :current
+      end
+    end
+
+    it_behaves_like 'authenticated_action'
+
+    describe 'should return the current user' do
+      before { action.call }
+
+      it { expect(JSON.parse(response.body)['user']['id']).to eq(user.id.to_s) }
+    end
+  end
+
+  describe 'POST #create' do
+    let(:action) do
+      proc do
+        post :create, user: { name: 'User 1' }
+      end
+    end
+
+    describe 'should return the current user' do
+      before do
+        request.env.delete 'HTTP_AUTHORIZATION'
```

Comment:
`request.env.delete 'HTTP_AUTHORIZATION'` this header shouldn't be used by anything. Could you see if it can be removed?